src/storage/garbage_collector.cpp

```diff
@@ -66,12 +97,15 @@ uint32_t GarbageCollector::ProcessUnlinkQueue() {
   }
 
   uint32_t txns_processed = 0;
-  // Certain transactions might not be yet safe to gc. Need to requeue them
   transaction::TransactionQueue requeue;
-  // It is sufficient to truncate each version chain once in a GC invocation because we only read the maximal safe
-  // timestamp once, and the version chain is sorted by timestamp. Here we keep a set of slots to truncate to avoid
-  // wasteful traversals of the version chain.
-  std::unordered_set<TupleSlot> visited_slots;
+
+  // Get active_txns in descending sorted order
+  std::vector<transaction::timestamp_t> active_txns = txn_manager_->GetActiveTxns();
```

Comment:
I would suggest that you declare 'active_txns' at line 90, so that readers can clearly see that 'active_txns' plays a similar role with the 'oldest_txn' in the regular GC. You can still assign the value of 'active_txns' here.